### DISCUSSION TITLE
Selecting files to download as an archive

### DISCUSSION THREAD

**Discussion by 6elkir** [rejetto/hfs#81] [Ideas]
It would be cool to be able to select some of them with checkmarks in the list of files and by clicking the "Download archive" button, download not the entire directory, but only the previously selected files. As for me - it will be very convenient.

**Comment by rejetto** [comment]
this feature is already available, but you have to click "filter list" button
<img width="373" alt="image" src="https://user-images.githubusercontent.com/1367199/202153924-57fa66f8-204f-4394-8f85-9962724da7be.png">

**Comment by rejetto** [comment]
I understand it is not clear that you have to click "filter list".
I made it so because I didn't want to clutter the list with the checkboxes all the time.

Maybe if extend this message
<img width="333" alt="image" src="https://user-images.githubusercontent.com/1367199/202156901-7fec8436-b46b-42be-9084-0bb46a07d4dd.png">

specifying that you can get only some files if you click "filter list", more people will find it.
Or keep same message but offer a third button "select some files".

Not a perfect solution, tho.

**Comment by 6elkir** [comment]
Greetings! In this case, I would recommend thinking towards an interface similar to the file manager of the same windows or similar from Linux. In addition, if you arrange the gui in the style of the OS of the current time, it will look like part of the OS, which will add even more ease of use. In the same win10, such options were removed in the roll-up ribbon

**Comment by rejetto** [comment]
there are several OSs, and several versions, so I don't see it as a feasible way.
In the end I opted by introducing a third button to guide the user
<img width="405" alt="image" src="https://user-images.githubusercontent.com/1367199/215893173-bfb3988a-8a28-4713-8c88-31ba930b91d3.png">

<img width="626" alt="image" src="https://user-images.githubusercontent.com/1367199/215893228-727e6813-dd22-490a-a159-a800a8e05c61.png">

and then you automatically get the checkboxes.
I think it's a decent solution

<img width="409" alt="image" src="https://user-images.githubusercontent.com/1367199/215893315-c9f20344-b865-4c0f-8a28-81e7de9b0df6.png">

**Reply by 6elkir** [reply]
I would also add a checkbox in the 1st line in the list header to mark / unmark all elements - so as not to pierce 98 out of 100 for example)

**Comment by rejetto** [comment]
the screenshot is cut and you were not seeing the "select all" button
<img width="600" alt="image" src="https://user-images.githubusercontent.com/1367199/215998840-c096a377-381c-428b-bda8-d5aa7a885ddf.png">

but now i wonder if people may miss it and search for the checkbox you are saying because they are used to it.
What you think?
I chose to put the buttons in the row with the filter field because
- there was more space than the other row
- i could put a full label "select all", and make it clear for those who are not used to the "checkbox on top"

I guess my options are now
1. leave it as it is
2. add the checkbox with simple standard behavior, and leave other buttons.
3. add the checkbox and remove the buttons "select all" and "invert". When you press it I can present you with a 3 choices in a dialog
- select all
- select all in this page
- invert selection

🤔

**Reply by 6elkir** [reply]
It seems to me that the flag is more familiar to people. inherits representation from standard file interfaces.

**Comment by rejetto** [comment]
it could be something like this
<img width="838" alt="image" src="https://user-images.githubusercontent.com/1367199/216845777-a089ebc1-f9aa-4911-8975-f1952b305356.png">

**Reply by 6elkir** [reply]
Nice!

**Reply by rejetto** [reply]
ok, published in 0.30.1
